terraform/modules/aws/waf_logging/kinesis-firehose.tf

```diff
@@ -1,5 +1,82 @@
-resource "aws_iam_role" "waf_firehose_iam_role" {
-    name = "waf-firehose-iam-role"
+resource "aws_kinesis_firehose_delivery_stream" "waf_kinesis_stream" {
+  provider    = aws.us
+  name        = "aws-waf-logs-govuk-pay-waf-kinesis-stream-${var.environment}"
+  destination = "splunk"
+
+  splunk_configuration {
+    hec_endpoint               = var.splunk_hec_endpoint
+    hec_token                  = var.splunk_hec_token
+    hec_acknowledgment_timeout = 600
+    hec_endpoint_type          = "Event"
+    s3_backup_mode             = "FailedEventsOnly"
+
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = aws_cloudwatch_log_group.waf_firehose_delivery_stream.name
+      log_stream_name = "SplunkDelivery"
+    }
+
+    processing_configuration {
+      enabled = "true"
+
+      processors {
+        type = "Lambda"
+
+        parameters {     
+            parameter_name  = "LambdaArn"
+            parameter_value = "${aws_lambda_function.kinesis_data_transformation_lambda.arn}:$LATEST"
+        }
+        parameters { 
+            parameter_name  = "BufferSizeInMBs"
+            parameter_value = 3
```

Comment:
any particular reason for this value (and one below)?

Reply:
Yes, the buffer size has to be below the maximum that a lambda can process. If the lambda is delivered a huge buffer it will fail. 60 seconds is the minimum, but is dynamically raised if delivery falls behind. It's just a mandatory property for some reason.

Reply:
I'll document this 😄